### PR TITLE
feat(logger): add support for MLflow artifact logging

### LINKED ISSUE #12: Generic logger for SegmentationLightningModule
### Proposal  

Introduce an "agnostic" logger, that uses composition to encapsulate one of a series of possible torch lightning loggers (e.g MLFlowLogger, TensorBoardLogger .. )

### Motivation and description

Currently, the code of  `SegmentationLightningModule` relies on tensorboard to log non-numerical values such as images. Since `LightningModule.log()` only supports scalar artifacts, to log anything else one must rely on the underlying `LightningModule.logger.experiment` [(docs)](https://lightning.ai/docs/pytorch/stable/extensions/logging.html#manual-logging-non-scalar-artifacts), and the `experiment` object has different function signatures depending on the logger used, it would be good to hide the logger-specific details into a class that exposes a common set of functionalities, such as `log_image(..)`. 

### Proposed solution

In my opinion, the agnostic logger should not extend `lightning.pytorch.loggers.logger.Logger`, like `MLFlowLogger` and `TensorBoardLogger`, but rather be a new object that encapsulates one of the existing loggers. Therefore, the `SegmentationLightningModule` will have two attributes to log: `SegmentationLightningModule.logger`,
 set by the trainer and used by `LightningModule.log()` and `SegmentationLightningModule.agnostic_logger`, used to log non-scalar values which relies on the underlying logger but hides the overhead of dealing with different function's signatures.

---

I appreciate any feedback on this suggestion. I am willing to contribute and proceed with a draft pull request after the review of this issue.

### PR DESCRIPTION
**Pull request by MicheleCattaneo**
### Description

This PR adds an `AgnosticLogger` as `self.agnostic_logger` to the `LightningModule` for segmentation.

### Purpose

To provide a unified interface for logging non-scalar values such as images, dataframes, and CSV files, with support for both TensorBoard and MLflow loggers.

As far as I know, PyTorch Lightning's built-in loggers do not share a consistent interface for logging non-scalar artifacts. Only `self.log()` for scalar values can be used safely across all logger types.

Why not extend `LightningLoggerBase` and override `self.logger`?
Unifying all backends (MLflow, TensorBoard, etc.) into a single logger class would be too complex, since each has its own API and internal workings. With this different type of class, instead, we can just add the custom behaviour that we need and keep the class very simple.

Provides convenience methods:

```python
log_img()
log_df()
log_params()
```

What do you think?

Closes #12

### CODE REVIEW

**Comment by colon3ltocard** [comment]
> Hi @colon3ltocard, yes it's clear what you mean, I think it's a good idea. Do you have any suggestion for the class names? Perhaps _agnostic_ is not the best name anymore in this case.
> 
> Also, thank you @tourniert for the feedback!
> 
> I will try to adapt the changes soon :)

What about these names : LoggerWrapperABC, MLFlowLoggerWrapper, TensorBoardLoggerWrapper. If it is too long remove "Logger" from all of them 😆

File: mfai/logging/lightning_logger_utils.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        is returned as numpy array.
+
+        Args:
+            img (torch.Tensor | np.ndarray): The image array
+
+        Returns:
+            np.ndarray: The image as numpy array
+        """
+        if hasattr(img, "detach"):
+            img = img.detach().cpu().numpy()
```
Comment (MicheleCattaneo) [inline mfai/logging/lightning_logger_utils.py:42]: Note for me: maybe this condition is not sufficient, there could be a tensor disconnected from the computational graph (no "detach") but still need to make it a np array

**Comment by colon3ltocard** [comment]
I close this PR since it is implemented by https://github.com/meteofrance/mfai/pull/142 and https://github.com/meteofrance/mfai/pull/141
